dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/EventManager.java

```diff
@@ -70,13 +71,16 @@
 public class EventManager
 {
     @NonNull
-    private final InsertValidatingEventChecker insertValidatingEventChecker;
+    @Qualifier( "checkersRunOnInsert" )
```

Comment:
This has not been fully communicated but within platform we have intentions to move away from using qualifiers as they interact poorly with other features like using lombok annotations to avoid reading/writing boilderplate. 
So maybe we can go a middle way and have lists but use a dedicated (sub-) type of `Checker` in each of the lists so they are unique by type? I like to define such narrow use subtypes as inner types of the parent type this way we don't get lots of files with basically no content and also pulling apart something that one rather would want to read as a unit.

Reply:
I heard that but do not fully know what I should do instead 😄 

I tried creating InsertChecker, UpdateChecker, DeleteChecker interfaces extending the Checker interface. Then have them injected based into the EventManager

https://github.com/dhis2/dhis2-core/pull/9366/commits/5b062c02f07a136c0acfdb4f586da8fddc920206

This works in theory but fails integration tests since Checker order matters. I could use the `@Order` annotation on the checkers, but am not happy with the root of it that Checkers are order-dependent.

Do you have a suggestion on how to make this work, while satisfying the platform team vision and not having to make Checkers order independent? 😄

Reply:
I made some changes based on your suggestions. My generics are a bit rusty 😂  So please have a look. I made sure that we get a compile-time error when for example declaring an InsertChecker to be run as part of the DeleteCheckers.